src/client/websocket/WebSocketManager.js

```diff
@@ -42,48 +63,81 @@ class WebSocketManager extends EventEmitter {
 
   /**
    * Destroy the client.
-   * @returns {void} Whether or not destruction was successful
+   * @returns {boolean} Whether or not destruction was successful
    */
   destroy() {
-    if (!this.connection) {
-      this.debug('Attempted to destroy WebSocket but no connection exists!');
+    if (!this.shards.size) {
+      this.debug('Attempted to destroy WebSocket but no connections exist!');
       return false;
     }
-    return this.connection.destroy();
+    this.packetManager.handleQueue();
+    for (const shard of this.shards.values()) shard.destroy();
+    return true;
   }
 
   /**
    * Send a packet on the available WebSocket.
    * @param {Object} packet Packet to send
-   * @returns {void}
+   * @returns {boolean}
    */
-  send(packet) {
-    if (!this.connection) {
-      this.debug('No connection to websocket');
-      return;
+  broadcast(packet) {
+    if (!this.shards.size) {
+      this.debug('No websocket connections');
+      return false;
     }
-    this.connection.send(packet);
+    for (const shard of this.shards.values()) shard.send(packet);
+    return true;
   }
 
   /**
-   * Connects the client to a gateway.
+   * Connects all appropriate shards to the gateway.
    * @param {string} gateway The gateway to connect to
-   * @returns {boolean}
+   * @param {Function} resolve Function to run when connection is successful
+   * @param {Function} reject Function to run when connection fails
    */
-  connect(gateway) {
-    if (!this.connection) {
-      this.connection = new WebSocketConnection(this, gateway);
-      return true;
-    }
-    switch (this.connection.status) {
-      case Status.IDLE:
-      case Status.DISCONNECTED:
-        this.connection.connect(gateway, 5500);
-        return true;
-      default:
-        this.debug(`Couldn't connect to ${gateway} as the websocket is at state ${this.connection.status}`);
-        return false;
-    }
+  spawn(gateway, resolve, reject) {
+    this.gateway = gateway;
+    const spawnShard = id => {
+      if (this.client.options.internalSharding && id >= this.client.options.shardCount) return;
+      this.debug(`Spawning shard ${id}`);
+      const shard = this.createShard(id);
+      shard.ws.once('error', reject);
+      shard.ws.once('close', event => {
+        if (event === 4004) reject(new Error('TOKEN_INVALID'));
```

Comment:
event.code

Reply:
I noticed it's not event.code anymore, I tested and this is the only way it responded. You can check for yourself, when I logged it `event` was a number, not an object.